starter-code/routes/drones.js

```diff
@@ -1,21 +1,37 @@
-const express = require('express');
+const express = require("express");
+const Drone = require("../models/drone");
 
 // require the Drone model here
 
 const router = express.Router();
 
-
-router.get('/drones', (req, res, next) => {
-  // Iteration #2
+router.get("/", (req, res, next) => {
+  Drone.find({}, (err, drones) => {
+    if (err) {
+      return next(err);
+    }
+    res.render("drones/index", { drones: drones });
+  });
 });
 
-
-router.get('/drones/new', (req, res, next) => {
-  // Iteration #3
+router.get("/new", (req, res, next) => {
+  res.render("drones/new");
 });
 
-router.post('/drones', (req, res, next) => {
-  // Iteration #3
+router.post("/", (req, res, next) => {
+  const droneInfo = {
+    name: req.body.name,
```

Comment:
By the way, you can use clever ES6 syntax to deconstruct the body object and and the shorthand object prop syntax to create the `droneInfo` object, like this:

`const {name, propellers, maxSpeed} = req.body; `
` const droneInfo = {name, propellers, maxSpeed}; `